codeforamerica/vita-min
# allow back link to to previous navigation actions in the current controller

## Link to pivotal/JIRA issue
- https://codeforamerica.atlassian.net/browse/FYST-1296
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- I added a branch to `StateFile::QuestionsController.prev_path` so that in case the current controller has multiple `navigation_actions` (think 1099-Gs, AZ public school contributions and AZ qualifying organization contributions) our app's back link will go to the first action in that list when clicked, instead of the previous behavior of jumping to the previous controller.
## How to test?
- Go through every possible flow for the unemployment, public school and qualifying org contribution pages and use the back link from every one of them to verify that it does a sensible thing.
- The only behavior with this change that I believe is slightly worse than the previous behavior is when you go to an empty index page (e.g. unemployment), submit without selecting a radio button and are shown a validation error, and then try to use the back link - it now shows you the same page again but without the validation error, because you're technically on the update action after submitting the blank radio buttons and the back link is bringing you back to the index action.
